### PR TITLE
feature(redshift): Adding way to define columns in schema()

### PR DESCRIPTION
**Pull request by fivetran-ankitsilla**
### Description of Change
1. Explicitly defining some columns and their type in schema()
2. Preserves the existing behaviour
3. Added filter field which will be used inside the where clause.

Jira: https://fivetran.atlassian.net/browse/RD-1229041

### Testing
Have added the testing details in the doc:
https://docs.google.com/document/d/1ZBwNOFxQ1YvayxQWlv3lbu-VyMgP6nTpEtGzQvh-xUE/edit?usp=sharing

### Checklist
Some tips and links to help validate your PR:

- [ ] Tested the connector with `fivetran debug` command.
- [ ] Added/Updated example-specific README.md file, see [the README template](https://github.com/fivetran/community_connectors/tree/main/_template_connector/README_template.md) for the required structure and guidelines.
- [ ] Followed Python Coding Standards, [refer here](https://github.com/fivetran/community_connectors/blob/main/PYTHON_CODING_STANDARDS.md)

### CODE REVIEW

File: redshift/large_data_volume/table_specs.py [hunk 1]
```diff
@@ -107,6 +118,19 @@
...
+            # creates them in the destination even when no non-null data is observed.
+            # Example: "quantity_sold": "SHORT", "commission": "FLOAT"
+        },
+        "filter": {
+            # Sync only records where saletime is after this date.
+            # Remove or set to None to sync all records.
+            "column": "saletime",
+            "operator": ">",
+            "value": "2008-01-01",
+        },
```
[the remark this answers is not in the document]
Comment (fivetran-ankitsilla) [reply redshift/large_data_volume/table_specs.py]: done

**Review by fivetran-JenasVimal — DISMISSED** [review]
LGTM , please ensure all the co-pilot comments have been resolved

File: redshift/large_data_volume/redshift_client.py [hunk 2]
```diff
@@ -281,7 +283,21 @@ def detect_typed_columns(selected_cols_with_types):
...
+    """
+    Build a parameterized static filter predicate from a TABLE_SPECS filter.
+    Args:
+        filter_condition: optional dict with keys 'column', 'operator', and 'value'
+    Returns:
+        A tuple of (where_condition, parameter), or (None, None) when no filter is configured.
+    """
+    if not filter_condition:
+        return None, None
+    return f'"{filter_condition["column"]}" {filter_condition["operator"]} %s', filter_condition["value"]
```
[the remark this answers is not in the document]
Comment (fivetran-ankitsilla) [reply redshift/large_data_volume/redshift_client.py]: added

**Review by fivetran-JenasVimal — APPROVED** [review]
LGTM

File: redshift/large_data_volume/table_specs.py [hunk 3]
```diff
@@ -107,6 +118,14 @@
...
         "include": [],
         "exclude": [],
+        "column_types": {
+            # Explicitly declare columns that may contain only NULL values so Fivetran
+            # creates them in the destination even when no non-null data is observed.
+            # Example: "quantity_sold": "SHORT", "commission": "FLOAT"
+        },
+        # Optional: apply a static WHERE condition to every sync of this table.
+        # Example:"filter": {"column": "saletime", "operator": ">", "value": "2008-01-01"}
+        "filter": None,
```
Comment (fivetran-sahilkhirwal) [inline redshift/large_data_volume/table_specs.py:128]: It would be great to have non empty values in these two fields to show they will look like when we fill value instead of None :)

**Review by fivetran-sahilkhirwal — APPROVED** [review]
Please do one more round of testing before merging covering all edge cases ( covering both full and incremental cases )
We should take special care because it is a promoted connector
